### PR TITLE
[KUBC-363] Added revert application for App-deployer

### PR DESCRIPTION
**Pull request by oumkale**
Signed-off-by: oumkale <oum.kale@mayadata.io>

**What this PR does / why we need it**:

Added revert application for Git-App-Deployer.
For sock-shop revert application has been added.
With ```operation``` flag.

**Special notes for your reviewer**:

### CODE REVIEW

File: custom/git-app-deployer/main.go [hunk 1]
```diff
@@ -62,31 +76,45 @@ func getKubeConfig() (*rest.Config, error) {
...
+		namespace: "",
+		filePath:  "",
+		timeout:   0,
+		operation: "",
+	}
+
+	appVars.namespace = *flag.String("namespace", "", "namespace for the application")
+	appVars.filePath = *flag.String("typeName", "weak", "type of the application")
+	appVars.timeout = *flag.Int("timeout", 300, "timeout for application status")
+	appVars.operation = *flag.String("operation", "apply", "revert application type")
```
Comment (ksatchit) [inline custom/git-app-deployer/main.go:90]: description can be changed

File: custom/git-app-deployer/main.go [hunk 2]
```diff
@@ -62,31 +76,45 @@ func getKubeConfig() (*rest.Config, error) {
...
 	nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
 	_, err := clientset.CoreV1().Namespaces().Create(nsSpec)
 	return err
 }
 
-//CreateSockShop creates sock-shop application
-func CreateSockShop(path string, ns string) error {
-	command := exec.Command("kubectl", "apply", "-f", path, "-n", ns)
+// DeleteNamespace delete a sock-shop/loadtest namespace
+func DeleteNamespace(clientset *kubernetes.Clientset, namespaceName string) error {
```
Comment (ksatchit) [inline custom/git-app-deployer/main.go:110]: This has the risk of getting "stuck" if there are residual resources. This needs to be invoked once all other resources are removed in provided ns

File: custom/git-app-deployer/main.go [hunk 3]
```diff
@@ -97,23 +125,87 @@ func CreateSockShop(path string, ns string) error {
...
 
+//CheckPodStatusForRevert checks the running application terminated or not
+func CheckPodStatusForRevert(appNs, appLabel string, timeout, delay int, operation string, clientset *kubernetes.Clientset) error {
+	err := retry.
+		Times(uint(timeout / delay)).
+		Wait(time.Duration(delay) * time.Second).
+		Try(func(attempt uint) error {
+			podSpec, err := clientset.CoreV1().Pods(appNs).List(metav1.ListOptions{LabelSelector: appLabel})
+			if err != nil {
+				return errors.Errorf("Unable to find the pods with matching labels, err: %v", err)
```
Comment (ksatchit) [inline custom/git-app-deployer/main.go:195]: not really labels..we are checking if there is no pod at all.

Comment (ksatchit) [reply custom/git-app-deployer/main.go:195]: We can add a check to see no chanengines exist here. If exists, log and bail out

File: custom/git-app-deployer/main.go [hunk 4]
```diff
@@ -17,9 +17,18 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
+//APPVars maintaining all parameters during application installation and uninstallation
+type APPVars struct {
```
Comment (ispeakc0de) [inline custom/git-app-deployer/main.go:21]: AppDetails

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**